pkg/cmd/cli/cmd/generate.go

```diff
@@ -109,3 +96,23 @@ $ s2i generate docker.io/centos/nodejs-10-centos7 Dockerfile.gen
 
 	return generateCmd
 }
+
+// manageConfigImageLabelsBuildImageName extracts the image labels from builder image in the provided config.
+// Returns an error if the builder image name is invalid or there is an error extracting the image labels.
+func manageConfigImageLabelsBuildImageName(ctx context.Context, cfg *api.Config) error {
+	builderImageName := CanonizeBuilderImageArg(cfg.BuilderImage)
+	ref, err := alltransports.ParseImageName(builderImageName)
+	if err != nil {
+		return err
+	}
+
+	cfg.BuilderImage = ref.DockerReference().Name()
+
+	var imageLabels map[string]string
+	if imageLabels, err = getImageLabels(ctx, ref); err != nil {
+		return err
+	}
+
+	cfg.BuilderImageLabels = imageLabels
```

Comment:
```suggestion
	if cfg.BuilderImageLabels, err = getImageLabels(ctx, ref); err != nil {
		return err
	}
```

I don't think there is a reason to create an additional variable here, should be able to assign directly to the future recipient.